File: python/src/remote_computation/logging/log_option.py
from common import ByteReader
from serialization import to_bytes


class LogOption:

    def __init__(self, reader: ByteReader):
        self.frequency = reader.read_int()
        self.log_last_n = reader.read_int()
        self.print = reader.read_bool()
        self.plot = reader.read_bool()
        self.min_max = reader.read_bool()
        self.running_average_smoothing = reader.read_float()

    def to_bytes(self) -> bytes:
        b = b''
        b += to_bytes(self.frequency)
        b += to_bytes(self.log_last_n)
        b += to_bytes(self.print)
        b += to_bytes(self.plot)
        b += to_bytes(self.min_max)
        b += to_bytes(self.running_average_smoothing)
        return b

    @staticmethod
    def create(frequency, log_last_n, print, plot, min_max, running_average_smoothing):
        b = b''

        b += to_bytes(frequency)
        b += to_bytes(log_last_n)
        b += to_bytes(print)
        b += to_bytes(plot)
        b += to_bytes(min_max)
        b += to_bytes(running_average_smoothing)

        return LogOption(ByteReader(b))

File: Assets/RemoteComputation/Logging/LogOption.cs
using System.Collections.Generic;
using Common;
using Common.ByteConversions;

namespace RemoteComputation.Logging
{
    public class LogOption : IByteConvertible
    {
        public int frequency;
        public int logLastN;
        public bool print;
        public bool plot;
        public bool minMax;
        public float runningAverageSmoothing;

        /// <summary> Creates a log option with the given name and parameters </summary>
        /// <param name="frequency"> How much epochs should elapse between the logging </param>
        /// <param name="logLastN">
        ///     Will only log the data for the last N epochs, if greater than zero. By default will log the data for all epochs
        /// </param>
        /// <param name="print"> Should the data be printed to the console? </param>
        /// <param name="plot"> Should the data be plotted to the graph? </param>
        /// <param name="minMax">
        ///     Should the graph also display an area for the minimum and maximum values for the epoch?
        /// </param>
        /// <param name="runningAverageSmoothing">
        ///     If greater than zero, will plot the running average of the data, with the given smoothing coefficient
        /// </param>
        public LogOption
        (int frequency = 1,
         int logLastN = 0,
         bool print = true,
         bool plot = true,
         bool minMax = true,
         float runningAverageSmoothing = 0f)
        {
            this.frequency               = frequency;
            this.logLastN                = logLastN;
            this.print                   = print;
            this.plot                    = plot;
            this.minMax                  = minMax;
            this.runningAverageSmoothing = runningAverageSmoothing;
        }

        /// <inheritdoc/>
        public IEnumerable<byte> Bytes =>
                frequency.ToBytes()
                         .ConcatMany(logLastN.ToBytes(),
                                     print.ToBytes(),
                                     plot.ToBytes(),
                                     minMax.ToBytes(),
                                     runningAverageSmoothing.ToBytes());
    }
}
